Mu2eKinKal/src/LoopHelixFit_module.cc

```diff
@@ -347,88 +475,86 @@ namespace mu2e {
       nseed += hseedcol.size();
       // loop over the seeds
       for(size_t iseed=0; iseed < hseedcol.size(); ++iseed) {
+        ++nSeen_;
         auto const& hseed = hseedcol[iseed];
+        if(!goodHelix(hseed)) continue; //determine if this seed should be fit
         auto hptr = HPtr(hseedcol_h,iseed);
-        // check helicity.  The test on the charge and helicity
-        if(hseed.status().hasAllProperties(goodseed_) ){
-          // test helix
-          auto const& helix = hseed.helix();
-          if(helix.radius() == 0.0 || helix.lambda() == 0.0 )
-            throw cet::exception("RECO")<<"mu2e::HelixFit: degenerate seed parameters" << endl;
-          auto zcent = Mu2eKinKal::zMid(hseed.hits());
-          // take the magnetic field at the helix center as nominal
-          VEC3 center(helix.centerx(), helix.centery(),zcent);
-          auto bnom = kkbf_->fieldVect(center);
-          // compute the charge from the helicity, fit direction, and BField direction
-          double bz = bnom.Z();
-          int charge = static_cast<int>(copysign(PDGcharge_,(-1)*helix.helicity().value()*fdir_.dzdt()*bz));
-          // test consistency.  Modify this later when the HelixSeed knows which direction it's going TODO
-          auto fitpart = fpart_;
-          if(charge*PDGcharge_ < 0){
-            if(usePDGCharge_)throw cet::exception("RECO")<<"mu2e::HelixFit: inconsistent charge" << endl;
-            fitpart = static_cast<PDGCode::type>(-1*fitpart); // reverse sign
+
+        // determine the fit direction hypothesis
+        auto helix_dir = chooseHelixDir(hseed);
```

Comment:
I recommend combining the 2 hseed functions into 1, returning a vector of fit directions. That could be empty (bad helix), have 1 entry, or have 2 entries. Then you don't have to worry about interactions between these functions.

Reply:
I'm not positive I follow this suggestion exactly. Is it that I return a vector of directions, loop over this filling a vector of track hypotheses, and then choose the best hypothesis from this list (which ideally should be length 1 or 2, where 0 means the helix seed was a bad seed)? If my understanding is right I can start making this change.

Reply:
>  If my understanding is right I can start making this change.
That's what I meant.
The goal is to make the code executed in produce as simple as possible

Reply:
I made this update, where I return a vector of length 0, 1, or 2 and loop over this vector and perform a track fit for each hypothesis. If ever extended, it will keep running `compare_tracks` taking the best result from the loop for any number of hypotheses (but of course there are only two at maximum as of now).